From 4f649e0a3e0f9ed1f0d6efdff5b14cdc40d84201 Mon Sep 17 00:00:00 2001
From: Jon Mason <jon.mason@arm.com
Date: Thu, 2 Mar 2023 15:22:08 +0000
Subject: [PATCH] Revert "vexpress64: pick DRAM size from DT"

This reverts commit 1a1143a45457161e90ea4cd5f3b0561d924ed8fe.

DRAM is determined via dtb in recent versions.  Since fvp isn't
reading and specifying a dtb, this fails and hangs u-boot.  Remove this
and go back to the way things were.

Signed-off-by: Jon Mason <jon.mason@arm.com>
Upstream-Status: Inappropriate
---
 board/armltd/vexpress64/vexpress64.c | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/board/armltd/vexpress64/vexpress64.c b/board/armltd/vexpress64/vexpress64.c
index af326dc6f453..e8ce88b22c5a 100644
--- a/board/armltd/vexpress64/vexpress64.c
+++ b/board/armltd/vexpress64/vexpress64.c
@@ -88,12 +88,20 @@ int board_init(void)
 
 int dram_init(void)
 {
-	return fdtdec_setup_mem_size_base();
+	gd->ram_size = PHYS_SDRAM_1_SIZE;
+	return 0;
 }
 
 int dram_init_banksize(void)
 {
-	return fdtdec_setup_memory_banksize();
+	gd->bd->bi_dram[0].start = PHYS_SDRAM_1;
+	gd->bd->bi_dram[0].size = PHYS_SDRAM_1_SIZE;
+#ifdef PHYS_SDRAM_2
+	gd->bd->bi_dram[1].start = PHYS_SDRAM_2;
+	gd->bd->bi_dram[1].size = PHYS_SDRAM_2_SIZE;
+#endif
+
+	return 0;
 }
 
 /* Assigned in lowlevel_init.S
